spec/requests/organizations/service_agreements_controller_spec.rb

```diff
@@ -132,4 +132,267 @@
+      let(:organization_administrator) { create :organization_member, organization: organization }
+
+      before { sign_in organization_administrator.user }
+
+      it 'successfully fetches the search results' do
+        perform_request
+        expect(response).to have_http_status(:success)
+        expect(JSON.parse(response.body, symbolize_names: true)).to eq({ results: search_result })
+        expect(response.body).to include 'Maria'
+        expect(response.body).not_to include 'Brigitte', 'Peter', 'Paul'
+      end
+
+      context 'when there is a search term' do
+        let(:search_term) { 'Brigitte' }
+
+        it 'successfully fetches the search results' do
+          perform_request
+          expect(response).to have_http_status(:success)
+          expect(JSON.parse(response.body, symbolize_names: true)).to eq({ results: search_result })
+          expect(response.body).not_to include 'Brigitte', 'Peter', 'Paul', 'Maria'
+        end
+      end
+    end
+
+    context 'when a civil servant is signed in' do
+      let(:civil_servant) { create :civil_servant, :full }
+
+      before { sign_in civil_servant.user }
+
+      it_behaves_like 'unauthorized json request' do
+        before { perform_request }
+      end
+    end
+
+    context 'when nobody is signed in' do
+      before { perform_request }
+
+      it_behaves_like 'unauthenticated json request' do
+        before { perform_request }
+      end
+    end
+  end
+
+  describe '#new' do
+    let(:perform_request) { get organizations_service_agreements_new_path(params: request_params) }
+    let(:organization) { create :organization }
+    let(:service_specification) { create :service_specification, organization: organization }
+
+    let(:maria) { create(:civil_servant, :full, first_name: 'Maria') }
+
+    let(:request_params) do
+      {
+        service_agreement: {
+          civil_servant_attributes: {
+            user_attributes: {
+              email: '-1'
+            }
+          }
+        }
+      }
+    end
+
+    context 'when a organization administrator is signed in' do
+      let(:organization_administrator) { create :organization_member, organization: organization }
+
+      before do
+        sign_in organization_administrator.user
+        perform_request
+      end
+
+      context 'when there is no civi set' do
+        it 'successfully renders the new service agreement form' do
+          expect(response).to render_template(:new)
+        end
+      end
+
+      context 'when there is a civi set' do
+        let(:request_params) do
+          {
+            service_agreement: {
+              civil_servant_attributes: {
+                user_attributes: {
+                  email: maria.user.email
+                }
+              }
+            }
+          }
+        end
+
+        it 'successfully renders the new service agreement form' do
+          expect(response).to render_template(:new)
+        end
```

Comment:
This spec is not different from the other, even though you have a different context. Maybe also expect that the form has the passed civil servant?